src/goto-symex/symex_dead.cpp

```diff
@@ -23,12 +23,9 @@ void goto_symext::symex_dead(statet &state)
 
   const codet &code=to_code(instruction.code);
 
-  if(code.operands().size()!=1)
-    throw "dead expects one operand";
-
-  if(code.op0().id()!=ID_symbol)
-    throw "dead expects symbol as first operand";
-
+  PRECONDITION(code.operands().size()!=1);
+	PRECONDITION(code.op0().id()!=ID_symbol);
```

Comment:
I don't know why there's this preceding space, and also the preconditions are wrong.

Something that is of the form

``` 
if (not something)
   throw exception
```

should have the form

```
assert (something);
```